File: src/main/java/五毒/第四周/第二遍/binary_tree_level_order_traversal.go
package 第二遍

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	result := [][]int{}
	if root == nil {
		return result
	}
	queue := []*TreeNode{root}

	for len(queue) > 0 {
		size := len(queue)
		var i int
		subList := make([]int, 0)
		for i = 0; i < size; i++ {
			curNode := queue[0]
			subList = append(subList, curNode.Val)
			queue = queue[1:]
			if curNode.Left != nil {
				queue = append(queue, curNode.Left)
			}
			if curNode.Right != nil {
				queue = append(queue, curNode.Right)
			}
		}
		result = append(result, subList)
	}
	return result
}

File: src/main/java/五毒/第四周/第二遍/binary_tree_level_order_traversal.java
package 五毒.第四周.第二遍;

import java.util.ArrayList;
import java.util.Deque;
import java.util.LinkedList;
import java.util.List;

public class binary_tree_level_order_traversal {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();
        if(root==null){
            return result;
        }
        Deque<TreeNode> queue = new LinkedList<>();
        queue.offerLast(root);
        while(!queue.isEmpty()){
            int size = queue.size();
            List<Integer> subList = new ArrayList<>();
            for(int i = 0;i<size;i++){
                TreeNode curNode = queue.removeFirst();
                subList.add(curNode.val);
                if(curNode.left!=null){
                    queue.offerLast(curNode.left);
                }
                if(curNode.right!=null){
                    queue.offerLast(curNode.right);
                }
            }
            result.add(new ArrayList<>(subList));
        }
        return result;
    }
}
